planex/cmd/createmockconfig.py

```diff
@@ -119,9 +119,9 @@ def parse_args_or_exit(argv=None):
                         help="mock config directory")
     parser.add_argument("-r", "--root", metavar="INCFG", required=True,
                         help="reference chroot config")
-    parser.add_argument("--enablerepo", action="append",
+    parser.add_argument("--enablerepo", action="append", default=['*'],
```

Comment:
This will mean that '*' is always enabled.   The default for `enablerepo` should be an empty list, the same as for `disablerepo` below.  Then in `main` you can call `load_yum_repos` as follows:
```
load_yum_repos(args.enablerepo or ['*'], args.disablerepo)
```

Reply:
I don't see how that is different. If enablerepo is supplied on the commandline it gets a value, if its not then it gets ['*'] both ways are equivalent in python AFAICT.

Reply:
`action="append"` means that the command line argument is appended to the default, so the enabled list will always begin with `'*'`

Reply:
However I'm not sure that enabling `'*'` when no argument is given is the right thing to do.   That will enable repositories which are disabled in the yum configuration.

Reply:
I think we should start with the set of repos which are enabled in the yum configuration, then remove disabled repos and add enabled ones:
```
base_set = set(yum.repos.listEnabled())
enabled_set = {repo for pattern in includes
               for repo in yum_base.repos.findRepos(pattern)}
disabled_set = {repo for pattern in excludes
                for repo in yum_base.repos.findRepos(pattern)}
return list ((base_set - disabled_set) | enabled_set)
```

This has the following properties (if ``--enablerepo` and `--disablerepo` both default to []):
 * If `--enablerepo` and `--disablerepo` are omitted, we get the set of repos which are enabled in the yum configuration
 * If `--enablerepo='*'` is given, we get all repos in the yum configuration
 * If `--enabledrepo='foo'` is given, we get all repos in the yum configuration plus foo
 * If `--enablerepo='foo'` and `--disablerepo='*'` are given, we get only foo

This matches my understanding of the semantics of `yum` and `dnf`'s `--enablerepos` and `--disablerepos` arguments.

@srowe You wrote this tool originally - what semantics did you expect?

Reply:
I use constructs like `--disablerepo='*' --enablerepo=foo` to enable only `foo`, I think we need to look at how `yum`/`dnf` process their options.

Reply:
I think that's how yum works, and my suggestion above should do the same (the last bullet - tested in the Python toplevel).    I guess we need to check whether yum accumulates all `--enablerepo` and `--disablerepo` arguments into two sets or does something which respects the order in which they appear.   If it's the latter we would need to change our argument parser to tag `--enablerepo` and `--disablerepo` arguments and accumulate them in a list which we then walk.   (I don't see any benefit in it behaving that way - it's a lot of complexity for limited benefit.)

Reply:
Order is significant

    [test@localhost ~]$ yum --disablerepo='*' --enablerepo=updates  repolist
    Loaded plugins: fastestmirror
    Repodata is over 2 weeks old. Install yum-cron? Or run: yum makecache fast
    Loading mirror speeds from cached hostfile
    repo id                                                     repo name                                                   status
    !updates                                                    updates                                                     1,241
    repolist: 1,241
    [test@localhost ~]$ yum --enablerepo=updates --disablerepo='*'  repolist
    Loaded plugins: fastestmirror
    Loading mirror speeds from cached hostfile
    repolist: 0

Reply:
Yes, I looked at the code for yum-cli and it processes the arguments in order and applies them directly to the yumbase object. (it doesn't use argparse)